ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateUpdater.java

```diff
@@ -284,47 +284,49 @@ public void commit() {
                             new BonsaiValue<>(wrappedWorldView().getCode(addr).orElse(null), null));
                 pendingCode.setUpdated(updatedAccount.getCode());
               }
+              if (!updatedAccount.getUpdatedStorage().isEmpty()) {
```

Comment:
nitpick: if we flip the condition and return early, we might avoid the indentions. i.e. `if (updatedAccount.getUpdatedStorage.isEmpty() {return;}`

Reply:
It's not clear to me why this improves withdrawals performance, can you please explain?

Also, it seems to be a change that will affect things other than withdrawals, so what has been done to ensure we haven't regressed anything else?

Finally, it's bonsai specific - what if we process withdrawals with FOREST?

Reply:
This looks like different optimisation from the one mentioned initially. I thought the optimization was to avoid the cost of the parallel streams overhead. This looks like it avoids doing unnecessary work if there is no storage to update. Which is a good thing though.

Are you planning to add an optimization to avoid the parallel stream for withdrawal balance updates?

Reply:
@siladu You're right, the PR description is not detailed enough. During block processing, we have 3 big consumers in terms of time processing : Transactions execution, commit the changes of each transaction to the updater and persist all the changes (in memory or on disk). During withdrawals processing, we also commit the changes of all the withdrawals to the worldstate updater but we use the same method (commit) that is used to manage all the cases (accounts deletion, storage update, storage deletion, ..etc). In the case of withdrawals, we only modify the balances, so we should just focus on accounts updates.
 @jframe good point, you're totally right. I wanted to start with this first optimisation to see if it is sufficient, and avoid creating another specific method for withdrawals to keep the code simple.

Reply:
I added the flamegraph at the block processing level to better see that withdrawals processing doesn't appear as it takes less time than before